### PR TITLE
Update for OpenSSL 3.0.0

### PR DESCRIPTION
**Pull request by straight-shoota**
This is a minimal patch to make our OpenSSL specs pass for the latest release OpenSSL 3.0.0 without errors and warnings. It does not necessarily add full support. Our test cases are a bit thin on some fronts, so they might not catch some behavioural changes. 
Nevertheless, this is a first step and getting it into nightly releases gives the opportunity to test the OpenSSL bindings in the wild. Make sure to install OpenSSL 3.0.0 for that.

Most of the relevant changes are just renamed methods but with consistent semantics.

A couple of algorithms have been deprecated (MD4 and RIPEMD160).

And some error messages have changed, which is relevant for our failure test which verify the specific error cause.

I also added explicit CI jobs for testing against 3.0 and 1.1.1 releases to validate compatibility.

Resolves https://github.com/crystal-lang/crystal/issues/11231

### CODE REVIEW

File: src/openssl/lib_crypto.cr [hunk 1]
```diff
@@ -174,8 +175,15 @@ lib LibCrypto
   fun evp_digestupdate = EVP_DigestUpdate(ctx : EVP_MD_CTX, data : UInt8*, count : LibC::SizeT) : Int32
   fun evp_md_ctx_copy = EVP_MD_CTX_copy(dst : EVP_MD_CTX, src : EVP_MD_CTX) : Int32
   fun evp_md_ctx_md = EVP_MD_CTX_md(ctx : EVP_MD_CTX) : EVP_MD
-  fun evp_md_size = EVP_MD_size(md : EVP_MD) : Int32
-  fun evp_md_block_size = EVP_MD_block_size(md : EVP_MD) : LibC::Int
+
+  {% if compare_versions(OPENSSL_VERSION, "3.0.0") >= 0 %}
```
Comment (Sija) [inline src/openssl/lib_crypto.cr:179]: This could use a feature check instead.

ditto elsewhere

Comment (straight-shoota) [reply src/openssl/lib_crypto.cr:179]: How so?

Comment (Sija) [reply src/openssl/lib_crypto.cr:179]: I meant checking for every new method signature via `LibCrypto.has_method?`, but perhaps, in this case, doesn't really make sense.

Comment (straight-shoota) [reply src/openssl/lib_crypto.cr:179]: This is the lib definiton. That's where these functions are defined.

Comment (Sija) [reply src/openssl/lib_crypto.cr:179]: image

**Comment by HertzDevil** [comment]
The Alpine package seems to have been renamed to `openssl3-dev` just right now: http://dl-cdn.alpinelinux.org/alpine/edge/main/x86_64/openssl3-dev-3.0.0-r3.apk

**Comment by straight-shoota** [comment]
Sigh. *CI can never be green for more than three days in a row.*

**Pull request merged**